src/main/scala/com/ing/wbaa/airlock/proxy/provider/LineageProviderAtlas.scala

```diff
@@ -1,147 +1,78 @@
-
-  private def fileEntities(
-      serverGuid: String,
-      bucketGuid: String,
-      bucketObject: String,
-      userSTS: String,
-      inputGuid: String,
-      inputType: String,
-      outputGuid: String,
-      outputType: String,
-      contentType: ContentType) = {
-    Entities(Seq(IngestedFile(
-      "DataFile",
-      userSTS,
-      FileAttributes(bucketObject, bucketObject, bucketObject, contentType.mediaType.value,
-        guidRef(bucketGuid, "Bucket"),
-        guidRef(serverGuid, "Server"),
-        List(guidRef(inputGuid, inputType)),
-        List(guidRef(outputGuid, outputType)),
-        Seq(Classification("customer_PII"))))))
-  }
+  def createLineageFromRequest(httpRequest: HttpRequest, userSTS: User): Future[LineageResponse] = {
 
-  private def processEntities(
-      serverGuid: String,
-      bucketGuid: String,
-      fileGuid: String,
-      userSTS: String,
-      method: String,
-      inputGuid: String,
-      inputType: String,
-      outputGuid: String,
-      outputType: String,
-      timestamp: Long) = {
-    Entities(Seq(Ingestion(
-      "aws_cli_script",
-      userSTS,
-      IngestionAttributes(s"aws_cli_${timestamp}", s"aws_cli_${timestamp}", method, userSTS,
-        guidRef(serverGuid, "Server"),
-        List(guidRef(inputGuid, inputType)),
-        List(guidRef(outputGuid, outputType))))))
-  }
+    val timestamp = System.currentTimeMillis()
+    val userName = userSTS.userName.value
+    val lh = getLineageHeaders(httpRequest)
+    val host = lh.host.getOrElse("unknown")
 
-  private def postEnities(userSTS: String, host: String, bucket: String, bucketObject: String, method: String, contentType: ContentType, timestamp: Long): Future[LineagePostGuidResponse] = {
-    for {
-      serverGuid <- postData(serverEntities(userSTS, host).toJson)
-      bucketGuid <- postData(bucketEntities(userSTS, bucket).toJson)
-      fileGuid <- method match {
-        case access if access == "read" =>
-          postData(
-            fileEntities(serverGuid, bucketGuid, bucketObject, userSTS, bucketGuid, "Bucket", serverGuid, "Server", contentType).toJson)
-        case access if access == "write" =>
-          postData(
-            fileEntities(serverGuid, bucketGuid, bucketObject, userSTS, serverGuid, "Server", bucketGuid, "Bucket", contentType).toJson)
-      }
-      processGuid <- method match {
-        case access if access == "read" =>
-          postData(
-            processEntities(serverGuid, bucketGuid, fileGuid, userSTS, method, bucketGuid, "Bucket", fileGuid, "DataFile", timestamp).toJson)
-        case access if access == "write" =>
-          postData(
-            processEntities(serverGuid, bucketGuid, fileGuid, userSTS, method, fileGuid, "DataFile", bucketGuid, "Bucket", timestamp).toJson)
-      }
-    } yield LineagePostGuidResponse(serverGuid, bucketGuid, fileGuid, processGuid)
-  }
+    def readOrWriteLineage(method: String, bucket: String): Future[LineagePostGuidResponse] = {
```

Comment:
I'd use the `AccesType` case class anywhere where this "method" is being used. Since we have this type already it ensures no mistakes with strings will be made